src/controllers/datalake.ts

```diff
@@ -0,0 +1,105 @@
+//     const fileSystemClient = serviceClient.getFileSystemClient(fileSystemName);
+//     const directoryClient = fileSystemClient.getDirectoryClient(directoryName);
+
+//     await directoryClient.create();
+// }
+
+export async function uploadFileToDataLake(fileName: string | undefined, fileContent: Buffer) {
+    logger.debug(`accountName: ${accountName}, accountKey: ${accountKey}, fileSystemName: ${fileSystemName}`);
+    if (fileName === undefined) {
+        throw new Error('File name is undefined');
+    }
+    if (fileContent === undefined) {
+        throw new Error('File name is undefined');
+    }
+    logger.debug(`Uploading file with file '${fileName}' to datalake`);
+    const fileSystemClient = serviceClient.getFileSystemClient(fileSystemName);
+    const directoryClient = fileSystemClient.getDirectoryClient(directoryName);
+    const fileClient = directoryClient.getFileClient(fileName);
+    const body = fileContent.toString();
+    await fileClient.create();
+    await fileClient.append(body, 0, fileContent.length);
+    await fileClient.flush(fileContent.length);
+}
+
+export async function deleteFileFromDataLake(fileName: string) {
+    const fileSystemClient = serviceClient.getFileSystemClient(fileSystemName);
+    const directoryClient = fileSystemClient.getDirectoryClient(directoryName);
+    const fileClient = directoryClient.getFileClient(fileName);
+
+    await fileClient.delete();
+}
+
+export async function listFilesInDirectory() {
+    const fileSystemClient = serviceClient.getFileSystemClient(fileSystemName);
+
+    const files = await fileSystemClient.listPaths({ path: directoryName });
+    const fileList = [];
+    for await (const file of files) {
+        if (file.name === undefined) {
+            continue;
+        }
+        logger.debug(`File: ${file.name}, isDirectory: ${file.isDirectory}`);
+        fileList.push({ name: basename(file.name), path: file.name, isDirectory: file.isDirectory });
+    }
+    return fileList;
+}
+
+export async function downloadFileFromDataLake(fileName: string) {
+    const fileSystemClient = serviceClient.getFileSystemClient(fileSystemName);
+    const directoryClient = fileSystemClient.getDirectoryClient(directoryName);
+    const fileClient = directoryClient.getFileClient(fileName);
+
+    const downloadResponse = await fileClient.read();
+    const body = downloadResponse.readableStreamBody;
+    if (body === undefined) {
+        throw new Error('ReadableStreamBody is undefined');
+    }
+
+    const downloaded = await streamToBuffer(body);
+
+    function streamToBuffer(readableStream: NodeJS.ReadableStream): Promise<Buffer> {
+        const chunks: Uint8Array[] = [];
+        if (readableStream === undefined) {
+            throw new Error('ReadableStream is undefined');
+        }
+        return new Promise((resolve, reject) => {
+            readableStream.on('data', (data) => {
+                chunks.push(data);
```

Comment:
Does this end up reading the entire file contents into core?
If so, it's probably acceptable for now but perhaps worth putting some comments in about how we can turn this into something where we write chunks straight into the Response body and flush them out to the client.
We'd need to set the Content-Length or use Transfer-Encoding: chunked so downloadFileFromDataLake would need to take a Response object that was ready to receive at least one more header and then a body.